src/client/dfuse/pil4dfs/aio.c

```diff
@@ -304,15 +315,20 @@ io_submit(io_context_t ctx, long nr, struct iocb *ios[])
 			DL_ERROR(rc, "daos_event_init() failed");
 			D_GOTO(err_loop, rc = daos_der2errno(rc));
 		}
-		d_iov_set(&iov, (void *)ios[i]->u.c.buf, ios[i]->u.c.nbytes);
-		sgl.sg_nr     = 1;
-		sgl.sg_iovs   = &iov;
 		ctx_ev->piocb = ios[i];
 		/* EQs are shared by contexts. Need to save ctx when polling EQs. */
 		ctx_ev->ctx = aio_ctx_obj;
+		D_ALLOC_PTR(req_args);
+		if (req_args == NULL)
+			D_GOTO(err_loop, rc = ENOMEM);
+		d_iov_set(&req_args->iov, (void *)ios[i]->u.c.buf, ios[i]->u.c.nbytes);
+		req_args->sgl.sg_nr     = 1;
+		req_args->sgl.sg_iovs   = &req_args->iov;
+
 		if (op == IO_CMD_PREAD) {
-			rc = dfs_read(d_file_list[fd]->dfs_mt->dfs, d_file_list[fd]->file, &sgl,
-				      ios[i]->u.c.offset, &read_size, &ctx_ev->ev);
+			daos_event_register_comp_cb(&ctx_ev->ev, aio_req_cb, req_args);
```

Comment:
this needs rc check

Reply:
ok. I will add rc check later. Thank you!